server/services/visitSessionsService.test.ts

```diff
@@ -530,19 +530,7 @@ describe('Visit sessions service', () => {
 
       expect(visitSchedulerApiClient.getVisitSessions).toHaveBeenCalledTimes(1)
       expect(results).toEqual(<VisitSlotList>{
-        'February 2022': [
-          {
-            date: 'Monday 14 February',
-            prisonerEvents: {
-              morning: [],
-              afternoon: [],
-            },
-            slots: {
-              morning: [],
-              afternoon: [],
-            },
-          },
-        ],
+        'February 2022': [],
```

Comment:
I'm thinking that if you can get it to just return `{}` for this case then in the template the existing `{% if slotsList | length %}` check would work and so no changes would need doing in the template itself 🧐?

Reply:
My logic being that we're not really doing what the test case says (i.e. "no results") for either the original or what you've changed it to. 

If trying this, probably need to check it does what's expected with multiple months: if, say, February has only a morning slot and March only an afternoon one that should return nothing for February but some data for March. Might need new test or tweak to an existing one.

Reply:
For info, there's also [dateAndTime.test.ts](https://github.com/ministryofjustice/book-a-prison-visit-staff-ui/blob/main/server/views/pages/dateAndTime.test.ts) where there are some tests for this template in isolation that might be handy.

Reply:
This whole issue arises because its difficult to reduce the data to {} in this case, do you think it's really important to try and get it to that stage?

Reply:
No, not necessarily - was just thinking of an alternative approach to this which would have been tidying the object returned from the service to remove empty months. E.g. in `visitSessionService.ts` around line 161 where you have:
```
if (visitSlotsforDay.slots.morning.length > 0 || visitSlotsforDay.slots.afternoon.length > 0) {
...
```
and an `else` block, something like:
```
        if (visitSlotsforDay.slots.morning.length > 0 || visitSlotsforDay.slots.afternoon.length > 0) {
          allVisitSlots.push(visitSlotsforDay)
          availableSessions[month] = allVisitSlots
        } else {
          delete availableSessions[month]
        }
      })
    })
```
Untested - and not saying this is particularly a better idea! This data's a bit fiddly however you do it...